ikarus/finiteelements/mechanics/enhancedassumedstrains.hh

```diff
@@ -171,10 +217,30 @@ protected:
         return Eigen::MatrixX<ScalarType>{};
     }();
 
+    auto strainFunction      = underlying().strainFunction(par, dx);
+    const auto C             = underlying().materialTangentFunction(par);
+    const auto geo           = underlying().localView().element().geometry();
+    const auto numberOfNodes = underlying().numberOfNodes();
+
     auto calculateMatrixContribution = [&]<typename EAST>(const EAST& easFunction) {
       typename EAST::DType D;
       calculateDAndLMatrix(easFunction, par, D, LMat, dx);
 
+      for (const auto& [gpIndex, gp] : strainFunction.viewOverIntegrationPoints()) {
+        const auto M            = easFunction.calcM(gp.position());
+        const double intElement = geo.integrationElement(gp.position()) * gp.weight();
+        const auto EVoigt       = strainFunction.evaluate(gpIndex, on(gridElement)).eval();
+        const auto CEval        = C(gpIndex);
```

Comment:
C should be evaluated for the current strain ÈVoigt` is this enforced here indirectlys?

Reply:
Yes. The existing interface of the [`materialTangentFunction`](https://github.com/ikarus-project/ikarus/blob/8e2263ed5ea6322412abcda9f37a3598023c6322/ikarus/finiteelements/mechanics/linearelastic.hh#L153) returns a lambda function, which takes `gpIndex` as the argument. This is now also implemented for `NonLinearElastic` (see [here](https://github.com/ikarus-project/ikarus/blob/572e568ba6707fd91055776c4623408f7d1e676e/ikarus/finiteelements/mechanics/nonlinearelastic.hh#L169)) where a similar interface is retained, and `C` is thereby evaluated for `Evoigt` indirectly. The interface for `materialTangentFunction` can also be modified such that it directly takes `Evoigt` as the argument.